float32 qOne
float32 qTwo
float32 qThree
